mods/mkrcraft/src/main/kotlin/pw/mkr/craft/blocks/binding/Binding.kt

```diff
@@ -0,0 +1,139 @@
+                val maximumBindings = 2 + (playTime * Utils.DAY_MULTIPLIER) + 0
+
+                if (playerBindings == maximumBindings) {
+                    destroyBlock()
+                    return sendMessage("You've already used (${playerBindings}/${maximumBindings}) bindings")
+                }
+
+                val binding = StoreManager.addBinding(
+                    Binding(
+                        blockChunk,
+                        pos.toModel(),
+                        itemStack.name.asString(),
+                        placer.name.asString()
+                    )
+                )
+
+                binding ?: return sendMessage("Failed to claim binding")
+
+                sendMessage("You have claimed ${binding.claimRadius} chunks around $blockChunk")
+                sendMessage("Bindings used: ${playerBindings + 1}/${maximumBindings}")
+            } else Utils.logger.error("Binding block placed by unknown entity!")
+        }
+    }
+
+    override fun onBreak(world: World?, pos: BlockPos?, state: BlockState?, player: PlayerEntity?) {
+        if (Utils.isServer) {
+            if (player != null && pos != null) {
+                val binding = StoreManager.chunkBoundTo(ChunkPos(pos).toModel())
+
+                // if this is a stray binding block that doesn't claim anything
+                binding ?: return super.onBreak(world, pos, state, player)
+
+                // this binding block is being broken by the claim owner
+                if (player.name.asString() == binding.player)
+                    return super.onBreak(world, pos, state, player)
+            }
+        } else super.onBreak(world, pos, state, player)
+    }
+
+    override fun afterBreak(
+        world: World?,
+        player: PlayerEntity?,
+        pos: BlockPos?,
+        state: BlockState?,
+        blockEntity: BlockEntity?,
+        stack: ItemStack?
+    ) {
+        super.afterBreak(world, player, pos, state, blockEntity, stack)
+
+        if (Utils.isServer && player != null) {
+            fun sendMessage(msg: String) = player.sendSystemMessage(LiteralText(msg), player.uuid)
+
+            val binding = StoreManager.chunkBoundTo(ChunkPos(pos).toModel())
+            binding ?: return
+
+            val removedBinding = StoreManager.removeBinding(binding)
+            removedBinding ?: sendMessage("Could not remove claim!")
+
+            sendMessage("Removed claim of ${binding.claimRadius} chunks around ${binding.chunk}")
+        }
+    }
+
+    companion object {
+        fun register() {
+            val bindingBlock =
+                BindingBlock(
+                    FabricBlockSettings.of(Material.METAL).strength(4.0f).requiresTool()
+                )
+            val bindingItem = BlockItem(bindingBlock, FabricItemSettings().group(ItemGroup.MISC))
+
+            Registry.register(Registry.BLOCK, Identifier("mkrcraft", "binding_block"), bindingBlock)
```

Comment:
Maybe we should move the variable parts of this to properties on the companion object, so when `PortalBlock` extends `BindingBlock`, we just need to override `strength` (`4.0f`), `name` (`binding_block`). We implicitly add `mkrcraft` as namespace.